oak-exercise/src/main/java/org/apache/jackrabbit/oak/exercise/security/authorization/models/simplifiedroles/Role.java

```diff
@@ -44,7 +49,9 @@ private Role(long permissions, String... privilegeNames) {
 
     private Role(@NotNull Role base, long permissions, String... privilegeNames) {
         this.permissions = base.permissions|permissions;
-        this.privilegeNames = ImmutableSet.<String>builder().addAll(base.privilegeNames).add(privilegeNames).build();
+        this.privilegeNames = Collections.unmodifiableSet(
+                (Set<String>) Stream.concat(base.privilegeNames.stream(), Arrays.stream(privilegeNames))
+                        .collect(Collectors.toCollection(LinkedHashSet::new)));
```

Comment:
There's a shortcut to create unmodifiable sets from Collectors class:

```suggestion
        this.privilegeNames = Stream.concat(base.privilegeNames.stream(), Arrays.stream(privilegeNames))
                .collect(Collectors.toUnmodifiableSet());
```

Reply:
This would not retain the original order to elements and that would change the code behaviour since ImmutableSet retains the original order.